ecosystem/typescript/sdk/src/plugins/ans_client.ts

```diff
@@ -94,6 +97,33 @@ export class AnsClient {
     return this.getAddressByDomainName(domain);
   }
 
+  /**
+   * Mint a new Aptos name
+   *
+   * @param account AptosAccount where collection will be created
+   * @param domainName  Aptos domain name to mint
+   * @param years year duration of the domain name
+   * @returns The hash of the transaction submitted to the API
+   */
+  async mintAptosName(account: AptosAccount, domainName: string, years: AnyNumber = 1): Promise<string> {
+    // check if the name is valid
+    if (domainName.match(nameComponentPattern) === null) {
+      throw new Error(`Name ${domainName} is not valid`);
+    }
+    // check if the name is available
+    const address = await this.getAddressByName(domainName);
+    if (address !== null) {
+      throw new Error(`Name ${domainName} is not available`);
+    }
+
+    const builder = new TransactionBuilderRemoteABI(this.provider.aptosClient, { sender: account.address() });
+    const rawTxn = await builder.build(`${this.contractAddress}::domains::register_domain`, [], [domainName, years]);
+
+    const bcsTxn = AptosClient.generateBCSTransaction(account, rawTxn);
+    const pendingTransaction = await this.provider.aptosClient.submitSignedBCSTransaction(bcsTxn);
+    return pendingTransaction.hash;
```

Comment:
As I was hacking on the Python SDK, it seems to make sense that the hash should probably be a special typed type and not just a string, even if it is just verified to be 32-bytes... or typed but not a string